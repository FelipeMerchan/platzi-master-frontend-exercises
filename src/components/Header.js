import React from 'react'
import './styles/Header.css'
import brandLogo from '../images/the-new-york-times-logotipo.svg'

function Header() {
  return (
      <header className="Header">
        <img className="Header__logotipo" width="600" src={brandLogo} alt="The New York Times" />
        <section className="subheading">
          <h2 className="text-left">Contenido de subcabecera</h2>
        </section>
      </header>
  )
}

export default Header