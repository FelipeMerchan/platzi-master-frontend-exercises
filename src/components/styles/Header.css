.Header {
  border-bottom: 2px solid var(--black-color);
}

.Header {
  padding: var(--space) 0;
  text-align: center;
}

.subheading {
  margin-bottom: var(--space);
  border-top: 2px solid var(--black-color);
}

.subheading h2 {
  font-family: var(--font-1);
  font-size: 1rem;
}