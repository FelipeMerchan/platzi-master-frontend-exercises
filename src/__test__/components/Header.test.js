import React from 'react'
import { mount } from 'enzyme'
import { create } from 'react-test-renderer'
import '../../setupTest'
import Header from '../../components/Header'

describe('Probar el componente <Header />', () => {
  const header = mount(<Header />)

  test('Render del componente Header', () => {
    expect(header.length).toEqual(1)
  })
  test('Render de la subcabecera', () => {
    expect(header.find('h2').text()).toEqual('Contenido de subcabecera')
  })
})

describe('Header Snapshot', () => {
  test('Comprobar la interfaz de usuario del componente Footer', () => {
    const header = create(<Header />)
    expect(header.toJSON()).toMatchSnapshot()
  })
})