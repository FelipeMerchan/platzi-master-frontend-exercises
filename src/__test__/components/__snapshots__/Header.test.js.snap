// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Snapshot Comprobar la interfaz de usuario del componente Footer 1`] = `
<header
  className="Header"
>
  <img
    alt="The New York Times"
    className="Header__logotipo"
    src="the-new-york-times-logotipo.svg"
    width="600"
  />
  <section
    className="subheading"
  >
    <h2
      className="text-left"
    >
      Contenido de subcabecera
    </h2>
  </section>
</header>
`;
